## PARCIAL ESTADÍSTICA

### Miguel Ángel Vélez Guerra - 1152469703

<hr/>

In [1]:
#-------Importing from other folder------#

import sys
sys.path.insert(0, "../resources/")

import mstats as ms

#-----------Miguel's statistics----------#

In [2]:
import scipy.stats as ss
import numpy as np

<hr/>

## 1. ANOVA 1 vía
Un estudio reciente comparó los salarios iniciales de los nuevos graduados en diversos campos. Los resultados se presentan a continuación. A un nivel del 5%. ¿Parece que hay diferencia en los salarios promedio de los graduados en los diferentes campos?

|Graduado|||Campos de estudio||
|---|
||Finanzas|Mercadeo|Sistemas|Economía|
|1|23.2|22.1|23.3|22.2|
|2|24.7|19.2|22.1|22.1|
|3|24.2|21.3|23.4|23.2|
|4|22.9|19.8|24.2|21.7|
|5|25.2|17.2|23.1|20.2|
|6|23.7|18.3|22.7|22.7|
|7|24.2|17.2|22.8|21.8|



In [3]:
finanzas = [23.2, 24.7, 24.2, 22.9, 25.2, 23.7, 24.2] # Resultados tratamiento 1
mercadeo = [22.1, 19.2, 21.3, 19.8, 17.2, 18.3, 17.2] # Resultados tratamiento 2
sistemas = [23.3, 22.1, 23.4, 24.2, 23.1, 22.7, 22.8] # Resultados tratamiento 3
economia = [22.2, 22.1, 23.2, 21.7, 20.2, 22.7, 21.8] # Resultados tratamiento 4

n_campos = len(finanzas + mercadeo + sistemas + economia) # Número total de observaciones
c_campos = 4 # Número total de tratamientos

alpha_campos = 0.05 # Nivel de significancia

- Se plantea la hipótesis nula y alternativa:

**Ho:** 𝜇finanzas = 𝜇mercadeo = 𝜇sistemas = 𝜇economia

**Ha:** 𝑛𝑜 𝑡𝑜𝑑𝑎𝑠 𝑙𝑎𝑠 𝑚𝑒𝑑𝑖𝑎𝑠 𝑠𝑜𝑛 𝑖𝑔𝑢𝑎𝑙𝑒𝑠

- Ahora encontraremos el valor crítico para rechazar la hipótesis nula

In [4]:
df1_campos = c_campos - 1
df2_campos = n_campos - c_campos
ms.hypothesis.crit_val_f(df1_campos, df2_campos, alpha_campos)

3.0087865704473615

- Ahora encontraremos el estadístico de prueba (RAZÓN F) para comparar con el valor crítico y determinar si se rechaza la hipótesis nula

In [5]:
ms.anova.get_fratio(finanzas, mercadeo, sistemas, economia)

20.688556457065562

Entonces, teniendo en cuenta que el valor de la RAZÓN F **20.688556457065562**, es **mayor** al valor crítico **3.0087865704473615**, se puede **rechazar con un nivel de significancia del *5%* la hipótesis nula.**

El estudio debe rechazar la hipótesis de que los salarios promedio de los graduados en los diferentes campos son iguales. Es decir, que estos salarios promedio **NO SON IGUALES** y **SI SE PRESENTAN DIFERENCIAS** entre ellos.

<hr/>

Para el ejercicio anterior vamos a hacer las pruebas para la diferencia entre pares de medias según el criterio de la diferencia mínima significativa (DMS), para encontrar cuales medias son iguales y cuales son diferentes

In [6]:
ms.anova.get_dms(finanzas, mercadeo, sistemas, economia, sign=alpha_campos)

|x_1 - x_2| = |24.014285714285712 - 19.3| = 4.714285714285712 > 1.3093138553397816
|x_1 - x_3| = |24.014285714285712 - 23.08571428571429| = 0.9285714285714235 <= 1.3093138553397816
|x_1 - x_4| = |24.014285714285712 - 21.985714285714288| = 2.028571428571425 > 1.3093138553397816
|x_2 - x_3| = |19.3 - 23.08571428571429| = 3.7857142857142883 > 1.3093138553397816
|x_2 - x_4| = |19.3 - 21.985714285714288| = 2.685714285714287 > 1.3093138553397816
|x_3 - x_4| = |23.08571428571429 - 21.985714285714288| = 1.1000000000000014 <= 1.3093138553397816


Podemos comprobar entonces, que de nuevo se afirma que los promedios de los salarios no son iguales, pues según el análisis DMS, algunos valores sobrepasan el valor DMS, refiriéndose a que los promedios son diferentes entre ellos.

Con esta información podemos decir que los campos de estudio que tienen en promedio **los mismos** salarios son:

- Finanzas y Sistemas
- Sistemas y Economía


También se puede decir que las zonas del país que pagan impuestos **diferentes** entre ellas son:

También se puede decir que los campos de estudio que tienen en promedio **diferentes** salarios son:

- Finanzas y Mercadeo
- Finanzas y Economía
- Mercadeo y Sistemas
- Mercadeo y Economía


<HR/>

## 2. Pruebas de hipótesis de 1 cola para la media poblacional en muestras pequeñas
Un nuevo bombillo producido por Sun Systems está diseñado para incrementar la vida útil de los bombillos a más de 5000 horas que es el promedio de los que actualmente existen. ¿El nuevo producto de Sun Systems proporciona una mejora si 25 bombillos se funden en promedio a las 5117 horas con s = 1886 horas? Fije alfa en 5%.

In [7]:
mu_sun = 5000 # Hipótesis nula de la media poblacional
n_sun = 25 # Tamaño de la muestra
x__sun = 5117 # Promedio muestral
s_sun = 1886 # Desviación estándar muestral

alpha_sun = 0.05 # Nivel de significancia

<u> **Paso 1**</u>: Plantear hipótesis

**Ho:** μ = 5000                  

**Ha:** μ > 5000

<u> **Paso 2**</u>: Nivel de significancia

In [8]:
alpha_sun

0.05

<u> **Paso 3**</u>: Valores críticos

In [9]:
df_sun = n_sun - 1
crit_sun = ms.hypothesis.crit_val_t(df_sun, alpha_sun, 'right')
crit_sun

1.7108820799094275

<u> **Paso 4**</u>: Estadístico de prueba (T)

In [10]:
t_sun = ms.generals.get_t(x__sun, mu_sun, s_sun, n_sun)
t_sun

0.31018027571580065

<u> **Paso 5**</u>: Decisión

In [11]:
ms.hypothesis.reject_h0(crit_sun, t_sun, 'right')

False

**NO se rechaza la hipótesis nula**, teniendo en cuenta que el estadístico de prueba *0.31018027571580065* no es mayor que el valor crítico *1.7108820799094275*.

<u>**Paso 6**</u>: Conclusión

Teniendo en cuenta un nivel de significancia del *5%*, podemos rechazar la hipótesis de que la duración promedio del nuevo bombillo producido por Sun Systems **es mayor** a 5000 horas.

<hr/>

## 3. Determinación del tamaño de la muestra para la media poblacional
Days Inn desea desarrollar un intervalo de confianza del 99% para estimar el número promedio de habitaciones ocupadas cada noche en sus localidades en toda la anción. ¿Cuántas noches deben incluirse en la muestra si se puede tolerar un error de 50 habitaciones y una muestra piloto revela que s = 165 habitaciones?.

In [12]:
alpha_days = 0.99 # Nivel de confianza
s_days = 165 # # Desviación estándar de la muestra piloto
error_days = 50 # Error tolerado

In [13]:
ms.intervals.n_avg(alpha_days, s_days, error_days)

72.0

**R/** Days Inn debería incluir **72** noches en la muestra para poder estimar con un nivel de confianza del *99%* y un error menor a *50*  habitaciones; el número promedio de habitaciones ocupadas cada noche en sus localidades en toda la nación.

<hr/>

## 4. Intervalos de confianza para la diferencia entre 2 medias poblacionales en muestras grandes
Dos procesos de producción se utilizan para producir tubos de acero. Una muestra de 100 tubos tomada del primer proceso de producción tiene una longitud en promedio de 27.3 pulgadas y s = 10.3 pulgadas. Las cifras correspondientes para los 100 tubos producidos por el segundo método son 30.1 y 5.2 . ¿Qué revela un intervalo del 99% sobre la diferencia de longitudes promedio de los tubos producidos por estos dos métodos?.

In [14]:
n_p1 = 100 # Tamaño de la muestra 1
x__p1 = 27.3 # Promedio de la muestra 1
s_p1 = 10.3 # Desviación estándar de la muestra 1

n_p2 = 100 # Tamaño de la muestra 2
x__p2 = 30.1 # Promedio de la muestra 2
s_p2 = 5.2 # Desviación estándar de la muestra 2

alpha_p = 0.99 # Nivel de confianza

In [15]:
ms.intervals.norm_2p(n_p1, n_p2, x__p1, x__p2, s_p1, s_p2, alpha_p)

(-5.7720427057731767, 0.17204270577317526)

**R/** A un nivel de confianza del *99%*, podemos observar que la diferencia entre la longitud promedio de los tubos producidos por estos 2 métodos está entre **-5.7720427057731767, 0.17204270577317526** pulgadas.

Entonces, como entre el intervalo de la diferencia está incluido el valor de **cero**, podemos concluir que la efectividad de estos 2 procesos de producción pueden ser iguales, ya que no se encuentran suficientes evidencias para demostrar que un proceso sea mejor que el otro.